tools/upip.py

```diff
@@ -134,6 +134,9 @@ def url_open(url):
             host, port = host.split(":")
             port = int(port)
         ai = usocket.getaddrinfo(host, port, 0, usocket.SOCK_STREAM)
+        ai = [_ai for _ai in ai if _ai[0] == usocket.AF_INET]
+        if len(ai) < 1:
+            fatal("No IPV4 address for host: '%s'" % host)
```

Comment:
This new code should probably go outside the try/except, because the try is intended to protect against `getaddrinfo` errors.

Also, because this code runs on very constrained systems would be good to optimise it to not use a list comprehension (which creates a new function to run the comprehension, and creates a temporary list).  Could instead be a for loop something this:
```python
for a in ai:
    if a[0] == AF_INET:
        ai = a
        break
else:
    fatal(...)
```

Reply:
Thanks @dpgeorge for the sugestions, I pushed a new commit to the branch and tested it before with the UNIX Port.

How do I handle the commit message warnings and code coverage issue from the automatic checks above?